base/linalg/symmetric.jl

```diff
@@ -138,7 +140,7 @@ A_mul_B!{T<:BlasComplex,S<:StridedMatrix}(C::StridedMatrix{T}, A::StridedMatrix{
 *(A::HermOrSym, B::HermOrSym) = full(A)*full(B)
 *(A::StridedMatrix, B::HermOrSym) = A*full(B)
 
-bkfact(A::HermOrSym) = bkfact(A.data, symbol(A.uplo), issymmetric(A))
+bkfact(A::HermOrSym) = bkfact(full(A.data), symbol(A.uplo), issymmetric(A))
```

Comment:
> We don't want to call full implicitly on sparse matrices since it might not fit in memory. It will be a big task to write a sparse Bunch-Kaufman factorization so we'll probably not be able to  support this method for a while. 

Is there some reference which I could look into to start working on this ?!

> Meanwhile, we could consider an error message suggesting `ldltfact` with a warning that 
> pivoting is only done as part of the symbolic factorization to reduce fill in.

I understand the synopsis that it should be an error message but I guess my knowledge is pretty limited and it would be really helpful if you point me to some sources on the understanding of this ! Thanks :-)